parser-typechecker/src/Unison/Util/SyntaxText.hs

```diff
@@ -1,25 +1,29 @@
+{-# LANGUAGE DeriveGeneric #-}
+
 module Unison.Util.SyntaxText where
 
 import Unison.Prelude
+import Unison.Name (Name)
 import Unison.Reference (Reference)
-import Unison.Referent (Referent)
+import Unison.Referent (TermRef)
 import Unison.HashQualified (HashQualified)
 import Unison.Pattern (SeqOp)
 
-import Unison.Util.AnnotatedText      ( AnnotatedText(..), annotate )
+import Unison.Util.AnnotatedText      ( AnnotatedText(..), annotate, segment)
 
-type SyntaxText = AnnotatedText Element
+type SyntaxText = SyntaxText' Reference
+type SyntaxText' r = AnnotatedText (Element r)
```

Comment:
Hehe, I think this is the same `'` generalization pattern that this PR tries to eliminate from `HashQualified'` and `Referent`.  I feel vindicated. :)

Reply:
I kind of hate it and it would be nice to have better names for these aliases. Maybe later.